rs/ic_os/metrics_tool/src/main.rs

```diff
@@ -0,0 +1,61 @@
+use anyhow::Result;
+use clap::Parser;
+
+use std::path::Path;
+use std::fs::File;
+use std::io::{self, BufRead};
+
+use ic_metrics_tool::{Metric, MetricsWriter};
+
+const INTERRUPT_FILTER: &str = "TLB shootdowns";
+const INTERRUPT_SOURCE: &str = "/proc/interrupts";
+const CUSTOM_METRICS_PROM: &str = "/run/node_exporter/collector_textfile/custom_metrics.prom";
+const TLB_SHOOTDOWN_METRIC_NAME: &str = "sum_tlb_shootdowns";
+
+#[derive(Parser)]
+struct MetricToolArgs {
+    #[arg(
+        short = 'm',
+        long = "metrics",
+        default_value = CUSTOM_METRICS_PROM
+    )]
+    /// Filename to write the prometheus metrics for node_exporter generation.
+    /// Fails badly if the directory doesn't exist.
+    metrics_filename: String,
+}
+
+fn get_sum_tlb_shootdowns() -> Result<u64> {
+    let path = Path::new(INTERRUPT_SOURCE);
+    let file = File::open(&path)?;
+    let reader = io::BufReader::new(file);
+
+    let mut total_tlb_shootdowns = 0;
+
+    for line in reader.lines() {
+        let line = line?;
+        if line.contains(INTERRUPT_FILTER) {
+            let parts: Vec<&str> = line.split_whitespace().collect();
```

Comment:
You don't need to collect if you already create an iterator from it on the next line.

`for part in line.split_whitespace().skip(1)` should work (or you can keep the `parts `variable if you prefer without `collect()`)

Reply:
Done